prov/rxm/src/rxm_cq.c

```diff
@@ -1364,7 +1362,7 @@ void rxm_ep_do_progress(struct util_ep *util_ep)
 		ret = rxm_msg_ep_recv(buf);
 		if (ret) {
 			if (OFI_LIKELY(ret == -FI_EAGAIN))
-				ofi_buf_free(&buf->hdr);
+				rxm_rx_buf_release(rxm_ep, buf, 0);
```

Comment:
If we know repost is false, calling ofi_buf_free() directly avoids the branch in rxm_rx_buf_release() (assuming the compiler doesn't just inline the code).